### PR TITLE
Add button to enable/disable user registrations

### PR DESCRIPTION
**Pull request by jfranciscosousa**
Why:

* We needed to disable/enable registrations on the admin dashboard

This change addresses the need by:
 
* Adding Settings table with methods to enable and disable registrations
* Add button to the Users index page on the dashboard, to toggle registrations

### CODE REVIEW

File: app/models/settings.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  validates :key, presence: true, uniqueness: true
+  validates :value, presence: true
+
+  def self.registrations_enabled?
+    Settings.find_by(key: :REGISTRATIONS_ENABLED)&.value == "true" || false
+  end
+
+  def self.toggle_registrations
+    registrations_enabled = Settings.find_by(key: :REGISTRATIONS_ENABLED)
+    registrations_enabled ||= Settings.create(key: :REGISTRATIONS_ENABLED, value: "true")
```
Comment (frm) [inline app/models/settings.rb]: Why not `find_or_initialize_by`?

```ruby
registrations_enabled = Settings.find_or_initialize_by(key: :registrations_enabled)
registrations_enabled.value = "true"
registrations_enabled.save
```

**Review by frm — CHANGES_REQUESTED** [review]

File: app/models/settings.rb (hunk 1 above)
Comment (jfranciscosousa) [reply app/models/settings.rb]: Rails API is big and my brain is not that big 😂 , I tried with create_by and had the error with the validations. This way seems perfect.

Comment (jfranciscosousa) [reply app/models/settings.rb]: Still had to verify if `value` existed before setting the default value.

File: spec/models/settings_spec.rb [hunk 2]
```diff
@@ -8,19 +8,27 @@
...
-  it "should toggle the registrations_enabled value for the first time" do
+  it "should activate registration if toggled for the first time" do
+    expect(Settings.toggle_registrations).to be(true)
+  end
+
+  it "should deactivate registrations if it's already activated" do
     Settings.toggle_registrations
 
-    expect(Settings.toggle_registrations).to be(true)
+    expect(Settings.toggle_registrations).to be(false)
```
Comment (frm) [inline spec/models/settings_spec.rb]: How about `expect(...).not_to be`?

File: spec/models/settings_spec.rb [hunk 3]
```diff
@@ -8,19 +8,27 @@
...
+    Settings.toggle_registrations
+
+    expect do
+      Settings.toggle_registrations
+    end.to change { Settings.count }.by(0)
   end
 
-  it "should toggle the registrations_enabled value for the first time" do
+  it "should activate registration if toggled for the first time" do
+    expect(Settings.toggle_registrations).to be(true)
```
Comment (frm) [inline spec/models/settings_spec.rb]: How about `expect(...).to be`?

File: spec/models/settings_spec.rb [hunk 4]
```diff
@@ -8,19 +8,27 @@
...
   end
 
-  it "should toggle the registrations_enabled value for the first time" do
-    expect(Settings.toggle_registrations).to be(false)
+  it "should not create additional settings record if toggled more than once" do
+    Settings.toggle_registrations
+
+    expect do
+      Settings.toggle_registrations
+    end.to change { Settings.count }.by(0)
```
Comment (frm) [inline spec/models/settings_spec.rb]: How about `expect(...).not_to change`?

File: app/models/settings.rb [hunk 5]
```diff
@@ -7,8 +7,9 @@ def self.registrations_enabled?
...
 
   def self.toggle_registrations
-    registrations_enabled = Settings.find_by(key: :REGISTRATIONS_ENABLED)
-    registrations_enabled ||= Settings.create(key: :REGISTRATIONS_ENABLED, value: "true")
+    registrations_enabled = Settings.find_or_initialize_by(key: :registrations_enabled)
+    registrations_enabled.value = "false" unless registrations_enabled.value
+    registrations_enabled.save
 
     registrations_enabled.value = registrations_enabled.value == "true" ? "false" : "true"
     registrations_enabled.save
```
Comment (frm) [inline app/models/settings.rb:18]: This should now be deleted, right?

Comment (jfranciscosousa) [reply app/models/settings.rb:18]: Nope.... because we store strings and the code at line 11 is to set the default value if it's `nil`.

Comment (jfranciscosousa) [reply app/models/settings.rb:18]: but yes we can only save it once

File: spec/models/settings_spec.rb (hunk 3 above)
Comment (jfranciscosousa) [reply spec/models/settings_spec.rb]: 👍

File: spec/models/settings_spec.rb (hunk 2 above)
Comment (jfranciscosousa) [reply spec/models/settings_spec.rb]: 👍

File: app/models/settings.rb (hunk 5 above)
Comment (frm) [reply app/models/settings.rb:18]: Why are we setting twice? We could do it with a `case` statement

```ruby
registrations_enabled.value = case registrations_enabled.value
  when nil, "false"
    "true"
  when "true"
    "false"
end
```

Comment (jfranciscosousa) [reply app/models/settings.rb:18]: Used to javascript, forgot that you could that with Ruby

Comment (frm) [reply app/models/settings.rb:18]: I know the feeling #elixir 😞

**Review by frm — APPROVED** [review]

**Pull request merged**